lib/auth/grpcserver.go

```diff
@@ -1135,6 +1135,73 @@ func (g *GRPCServer) DeleteAllDatabaseServers(ctx context.Context, req *proto.De
 	return &emptypb.Empty{}, nil
 }
 
+// GetAllDatabaseServices returns all registered DatabaseServices.
+func (g *GRPCServer) GetAllDatabaseServices(ctx context.Context, _ *proto.GetAllDatabaseServicesRequest) (*proto.DatabaseServiceV1List, error) {
+	auth, err := g.authenticate(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	dbs, err := auth.GetAllDatabaseServices(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	ret := &proto.DatabaseServiceV1List{
+		Services: make([]*types.DatabaseServiceV1, len(dbs)),
+	}
+	for i, db := range dbs {
+		dbServiceV1, ok := db.(*types.DatabaseServiceV1)
+		if !ok {
+			return nil, trace.BadParameter("unexpected DatabaseService type: %T", db)
+		}
+		ret.Services[i] = dbServiceV1
```

Comment:
In case of  `if !ok` condition the `i` service element will be nil. I would suggest to init array with  capacity like `make([]*types.DatabaseServiceV1 0, , len(dbs))` and append elements instead of creating a pre-initlized  array with `len(dbs)` size.

Reply:
In case of `!ok` it will end the method and return immediately
So, in the happy path there's no nil services in `ret.Services`
If an error occurs, this is GC'd and the allocation doesn't matter.

Anyway, this might be removed as suggested here:
https://github.com/gravitational/teleport/pull/19469#discussion_r1052248625

I'll make sure to not backport this PR before we resolve this comment

Reply:
`GetAllDatabaseServices` removed
https://github.com/gravitational/teleport/pull/19469#discussion_r1052444877